front-end/src/app/shared/services/FecApiService/fec-api.service.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { FecApiService } from './fec-api.service';
+import { FecApiPaginatedResponse } from 'app/shared/models/fec-api.model';
+import { CommitteeAccount } from 'app/shared/models/committee-account.model';
+import { environment } from 'environments/environment';
+
+describe('FecApiService', () => {
+  let httpTestingController: HttpTestingController;
+  let service: FecApiService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    httpTestingController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(FecApiService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+  
+  describe('#getDetails()', () => {
+    it('should return committee details', () => {
+      const committeeAccount: CommitteeAccount = new CommitteeAccount();
+      const response: FecApiPaginatedResponse = {
+        "api_version": "1.0",
+        "pagination": {
+            "page": 1,
+            "per_page": 20,
+            "count": 1,
+            "pages": 1
+        },
+        results: [committeeAccount]
+      }
+
+      service.getDetails("C00601211")
+        .subscribe(committeeAccountData => {
+          expect(committeeAccountData).toEqual(committeeAccount);
+        });
```

Comment:
So the service call to the FEC endpoint is being intercepted and mocked, yes?

Reply:
not actually calling FEC. httpTestingController tells the test if that url is requested (test part 1) and returns the mock response which line 40 checks (test part 2)